test/unittest/dds/conditions/ConditionsTests.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <fastdds/dds/core/conditions/GuardCondition.hpp>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+TEST(ConsitionsTests, GuardConditionActivation)
+{
+    GuardCondition cond;
+    ASSERT_FALSE(cond.get_trigger_value());
+    ASSERT_EQ(cond.set_trigger_value(true), ReturnCode_t::RETCODE_OK);
```

Comment:
```suggestion
   cond.set_trigger_value(true);
```
It will always return RETCODE_OK so I think that it is not necessary to assert it. Same for line 30